pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -55,7 +55,7 @@ in
 , changelog ? null
 , mainProgram ? null
 , doCoverage ? false
-, doHaddock ? !(ghc.isHaLVM or false)
+, doHaddock ? !(ghc.isHaLVM or false) && ghc.hasHaddock
```

Comment:
I think this suggestion makes sense.  But on the other hand, as an end user, if I pass `doHaddock = true`, I'd expect either I get haddocks when the build finishes, or the build fails.  It could be somewhat confusing to specify `doHaddock = true`, but have the build successfully finish without building haddocks.

Either way, I'd say this potential fix probably shouldn't hold up merging in this PR.